graphql/resolvers.go

```diff
@@ -893,6 +894,18 @@ func (r *taskResolver) BaseTaskMetadata(ctx context.Context, at *restModel.APITa
 	return &baseTaskMetadata, nil
 }
 
+func (r *taskResolver) SpawnHostLink(ctx context.Context, at *restModel.APITask) (*string, error) {
+	host, err := host.FindOne(host.ById(*at.HostId))
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("error finding host for task %s", *at.Id))
+	}
+	if host == nil || host.Distro.Provider == evergreen.HostTypeStatic || host.Distro.Provider == evergreen.ProviderNameDocker || host.Distro.SpawnAllowed == false {
```

Comment:
For the providers, you can check if they are in the list evergreen.ProviderSpawnable